src/main/java/server/Server.java

```diff
@@ -0,0 +1,134 @@
+                ClientHandler clientHandler = new ClientHandler(clientSocket);
+                clients.add(clientHandler);
+                clientHandler.start();
+                broadcastMessage("[SERVER] " + clientHandler.getClientName() + " connected successfully.");
+            }
+        } catch (IOException e) {
+            Logger logger = Logger.getLogger(Server.class.getName());
+            logger.log(Level.SEVERE, "Error occurred while starting the server.", e);
+        }
+    }
+
+    public void stop() {
+        try {
+            serverSocket.close();
+            for (ClientHandler client : clients) {
+                client.disconnect();
+            }
+        } catch (IOException e) {
+            Logger logger = Logger.getLogger(Server.class.getName());
+            logger.log(Level.SEVERE, "Error occurred while stopping the server.", e);
+        }
+    }
+
+    private void broadcastMessage(String message, ClientHandler client) {
+        client.sendMessage(message);
+    }
+
+    private class ClientHandler extends Thread {
+        private Socket clientSocket;
+        private String clientName;
+        private LocalDateTime connectionTime;
+        private BufferedReader reader;
+        private PrintWriter writer;
+
+        public ClientHandler(Socket clientSocket) {
+            this.clientSocket = clientSocket;
+            this.clientName = "client-" + (clients.size() + 1);
+            this.connectionTime = LocalDateTime.now();
+        }
+
+        public String getClientName() {
+            return clientName;
+        }
+
+        public void run() {
+            try {
+                reader = new BufferedReader(new InputStreamReader(clientSocket.getInputStream()));
+                writer = new PrintWriter(clientSocket.getOutputStream(), true);
+
+                String inputLine;
+                while ((inputLine = reader.readLine()) != null) {
+                    if (inputLine.equals(EXIT_COMMAND)) {
+                        break;
+                    } else if (inputLine.startsWith(FILE_COMMAND)) {
+                        handleFileCommand(inputLine);
+                    }
+                }
+            } catch (IOException e) {
+                Logger logger = Logger.getLogger(ClientHandler.class.getName());
+                logger.log(Level.SEVERE, "Error occurred while running the client handler.", e);
+            } finally {
+                disconnect();
+                clients.remove(this);
+                broadcastMessage("[SERVER] " + clientName + " disconnected.", this);
+            }
+        }
+
+        public void sendMessage(String message) {
+            writer.println(message);
+        }
+
+        public void disconnect() {
+            try {
+                reader.close();
+                writer.close();
+                clientSocket.close();
+            } catch (IOException e) {
+                Logger logger = Logger.getLogger(Client.class.getName());
```

Comment:
would be better to get the logger on the class level and then reuse it in methods